File: sdks/swift/Sources/PuxbaySDK/Models/Order.swift
import Foundation

public struct Order: Codable {
    public let id: String
    public let orderNumber: String
    public let status: String
    public let subtotal: Double
    public let taxAmount: Double
    public let totalAmount: Double
    public let amountPaid: Double
    public let paymentMethod: String
    public let orderingType: String
    public let offlineUuid: String?
    public let customer: String?
    public let customerName: String?
    public let cashier: String?
    public let cashierName: String?
    public let branch: String
    public let branchName: String
    public let items: [OrderItem]?
    public let createdAt: String
    public let updatedAt: String
}

public struct OrderItem: Codable {
    public let id: String
    public let product: String
    public let productName: String
    public let sku: String
    public let itemNumber: String
    public let quantity: Int
    public let price: Double
    public let costPrice: Double?
    public let getTotalItemPrice: Double? // Computed field in API maybe?
}

File: sdks/rust/src/models/order.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Order {
    pub id: String,
    pub order_number: String,
    pub status: String,
    pub subtotal: f64,
    pub tax_amount: f64,
    pub total_amount: f64,
    pub amount_paid: f64,
    pub payment_method: String,
    pub ordering_type: String,
    pub offline_uuid: Option<String>,
    pub customer: Option<String>,
    pub customer_name: Option<String>,
    pub cashier: Option<String>,
    pub cashier_name: Option<String>,
    pub branch: String,
    pub branch_name: String,
    #[serde(default)]
    pub items: Vec<OrderItem>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct OrderItem {
    pub id: String,
    pub product: String,
    pub product_name: String,
    pub sku: String,
    pub item_number: String,
    pub quantity: i32,
    pub price: f64,
    pub cost_price: Option<f64>,
    pub total_price: f64,
}
